### PR TITLE
feat(FTL-13450): add pagination to list-users-in-group command

### PR DESCRIPTION
**Pull request by YoussefAWasfy**
These recordings are before sorting was implemented:

https://github.com/affinidi/affinidi-cli/assets/112941926/eb7b6989-077d-4b49-b72b-eb4ad022ea01

https://github.com/affinidi/affinidi-cli/assets/112941926/e31fba6d-42b4-445c-8cf5-b5072cdccdb7

### CODE REVIEW

File: src/services/affinidi/vp-adapter/index.ts [hunk 1]
```diff
@@ -73,7 +73,8 @@ class VPAdapterService {
   public listLoginConfigurations = async (): Promise<ListLoginConfigurationOutput> => {
     const headers = await getBFFHeaders()
     try {
-      const response = await this.client.v1.listLoginConfigurations({ headers })
+      // TODO: implemet pagination
```
Comment (pulkitb2) [inline src/services/affinidi/vp-adapter/index.ts:76]: Can we remove the TODO

File: src/commands/login/list-users-in-group.ts [hunk 2]
```diff
@@ -13,21 +14,43 @@ export class ListUsersInGroup extends BaseCommand<typeof ListUsersInGroup> {
     'group-name': Flags.string({
       summary: 'Name of the user group',
     }),
+    'max-items': Flags.integer({
+      summary: "The total number of items to return in the command's output",
+      max: MAX_ITEMS_LIMIT,
```
Comment (carlos-affinidi) [inline src/commands/login/list-users-in-group.ts:19]: Should the max limit be defined in the CLI? Seems more of a restriction that should be in place in the API itself.
Or are you configuring the default? In which case we should just use the APIs default (not set it) unless that doesn't work for some reason.

File: src/commands/login/list-users-in-group.ts [hunk 3]
```diff
@@ -13,21 +14,43 @@ export class ListUsersInGroup extends BaseCommand<typeof ListUsersInGroup> {
...
-    const listGroupUsersOutput = await vpAdapterService.listGroupUsers(validatedFlags['group-name'])
+    const listGroupUsersOutput = await vpAdapterService.listGroupUsers(validatedFlags['group-name'], {
+      limit: validatedFlags['max-items'] ?? MAX_ITEMS_LIMIT,
+      exclusiveStartKey: validatedFlags['starting-token'] ?? undefined,
+    })
     ux.action.stop('Fetched successfully!')
+    const { lastEvaluatedKey, ...rest } = listGroupUsersOutput
+    if (!this.jsonEnabled()) this.logJson({ ...rest, 'starting-token': lastEvaluatedKey })
+
+    if (listGroupUsersOutput.lastEvaluatedKey && (await confirm({ message: 'Do you want to fetch next page?' }))) {
```
Comment (carlos-affinidi) [inline src/commands/login/list-users-in-group.ts:45]: Check if `flags['no-input']` is set before prompting. And as discussed with Digvijay we may want to use inquirer to jump ahead, behind or exit.

**Review by carlos-affinidi — CHANGES_REQUESTED** [review]

File: src/commands/login/list-users-in-group.ts (hunk 2 above)
Comment (YoussefAWasfy) [reply src/commands/login/list-users-in-group.ts]: I removed the limitation of the flag, but I set the default to 15.

Comment (YoussefAWasfy) [reply src/commands/login/list-users-in-group.ts]: The limit set in the API is 100

**Review by carlos-affinidi — CHANGES_REQUESTED** [review]

File: src/commands/login/list-users-in-group.ts [hunk 4]
```diff
@@ -6,28 +7,106 @@ import { INPUT_LIMIT } from '../../common/validators'
...
+    if (!this.jsonEnabled())
+      this.logJson({
+        ...rest,
+        'starting-token': lastEvaluatedKey,
+      })
+
+    if (
+      (lastEvaluatedKey || ListUsersInGroup.lastEvalutatedKeys) &&
+      !flags['no-input'] &&
+      !(startingToken && ListUsersInGroup.pageNumber === 1)
```
[the remark this answers is not in the document]
Comment (YoussefAWasfy) [reply src/commands/login/list-users-in-group.ts:61]: We are not showing prompt if starting token is provided

[the remark this answers is not in the document]
Comment (carlos-affinidi) [reply src/commands/login/list-users-in-group.ts:61]: Sorry, I wasn't clear. My question is if we can do it, or is there a reason it is disabled?

[the remark this answers is not in the document]
Comment (YoussefAWasfy) [reply src/commands/login/list-users-in-group.ts:61]: Yes, the reason is that if someone provided the token, then we won't be able to calculate the current page number. That was discussed with Digvijay and it was decided to not allow it.

**Review by carlos-affinidi — APPROVED** [review]

**Pull request merged**